### PR TITLE
Fixes #13664 - rescue smtp errors if mail failed on host built

### PR DESCRIPTION
**Pull request by shlomizadok**

### CODE REVIEW

**Comment by shlomizadok** [comment]
[test]

**Comment by shlomizadok** [comment]
[test]

**Comment by shlomizadok** [comment]
[test]

File: app/models/host/managed.rb [hunk 1]
```diff
@@ -1055,5 +1055,7 @@ def extract_params_from_object_ancestors(object)
   def send_built_notification
     recipients = owner ? owner.recipients_for(:host_built) : []
     MailNotification[:host_built].deliver(self, :users => recipients) if recipients.present?
+  rescue SocketError, Net::SMTPAuthenticationError, Net::SMTPServerBusy, Net::SMTPSyntaxError, Net::SMTPFatalError, Net::SMTPUnknownError => e
```
Comment (domcleal) [inline app/models/host/managed.rb]: or is `Net::SMTPError` enough?

File: app/models/host/managed.rb [hunk 2]
```diff
@@ -1055,5 +1055,7 @@ def extract_params_from_object_ancestors(object)
...
+    logger.debug "Host has been created. Failed to send email with #{e}"
```
Comment (domcleal) [inline app/models/host/managed.rb]: 1. Should be at least warning level, nobody will know it's a problem if at debug.
2. Use the exception logger.

**Comment by domcleal** [comment]
If you're getting intermittent integration test failures, just ignore them and/or comment rather than re-test.  It ties up valuable resources from other people testing PRs.

File: app/models/host/managed.rb (hunk 2 above)
Comment (shlomizadok) [inline app/models/host/managed.rb]: What is exception logger?

**Comment by shlomizadok** [comment]
Cool. Next time I'll do %w(test) :see_no_evil:

File: app/models/host/managed.rb (hunk 2 above)
Comment (domcleal) [inline app/models/host/managed.rb]: Foreman::Logging.exception, it accepts a logging level etc.

**Comment by shlomizadok** [comment]
Failure unrelated. Shall I have a pregnancy test?
(I have to write the word test... :smile: )

File: app/models/host/managed.rb [hunk 3]
```diff
@@ -1055,5 +1055,7 @@ def extract_params_from_object_ancestors(object)
   def send_built_notification
     recipients = owner ? owner.recipients_for(:host_built) : []
     MailNotification[:host_built].deliver(self, :users => recipients) if recipients.present?
+  rescue SocketError, Net::SMTPError => e
+    Foreman::Logging.exception(_("Host has been created. Failed to send email"), e)
```
Comment (domcleal) [inline app/models/host/managed.rb]: This string should be untranslated, all logs are in English only.

**Comment by shlomizadok** [comment]
Removed translations

**Comment by domcleal** [comment]
Merged as a90e4f2, thanks @shlomizadok.

**Pull request closed without merging**